File: SIMPLEAPI_Demo/SIMPLEAPI_Demo/Impresion/Core/Classes/PrinterWork.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SIMPLEAPI_Demo.Impresion.Core.Classes
{
    public class PrinterWork
    {
        public bool IsTestWork { get; set; }

        public List<PrinterRow> Rows { get; private set; }

        public PrinterWork()
        {
            Rows = new List<PrinterRow>();
        }

        public void AddRow(PrinterRow row)
        {
            Rows.Add(row);
        }


        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();

            foreach (var row in Rows)
                string.Join("||",
                    row
                    .Columns
                    .Select(x =>
                        x is PrinterTextColumn ? (x as PrinterTextColumn).Text :
                        x is PrinterImageColumn ? "{ Image }" :
                        x.GetType().Name
                    ));

            return sb.ToString();
        }

        public List<string> ToLines()
        {
            var Lines = new List<string>();
            foreach (var row in Rows)
                Lines.AddRange(row.ToLines());

            //var linesStrng = string.Join(Environment.NewLine, Lines);

            return Lines;
        }
    }
}

File: SIMPLEAPI_Demo/SIMPLEAPI_Demo_VB/Impresion/Core/Classes/printerWork.vb
Imports System.Text

Public Class printerWork
    Public Property IsTestWork As Boolean
    Public Property Rows As List(Of PrinterRow)

    Public Sub New()
        Rows = New List(Of PrinterRow)()
    End Sub

    Public Sub AddRow(ByVal row As PrinterRow)
        Rows.Add(row)
    End Sub

    Public Overrides Function ToString() As String
        Dim sb As StringBuilder = New StringBuilder()

        For Each row In Rows
            String.Join("||", row.Columns.[Select](Function(x) If(TypeOf x Is PrinterTextColumn, (TryCast(x, PrinterTextColumn)).Text, If(TypeOf x Is PrinterImageColumn, "{ Image }", x.[GetType]().Name))))
        Next

        Return sb.ToString()
    End Function

    Public Function ToLines() As List(Of String)
        Dim Lines = New List(Of String)()

        For Each row In Rows
            Lines.AddRange(row.ToLines())
        Next
        '//var linesStrng = string.Join(Environment.NewLine, Lines);
        Return Lines
    End Function
End Class
